packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2032,8 +2030,10 @@ export class ContainerRuntime
 				try {
 					checkpoint.rollback((message: BatchMessage) =>
 						this.rollback(
-							message.deserializedContent.type,
-							message.deserializedContent.contents,
+							message.type,
+							message.contents === undefined
```

Comment:
Is this back-compat logic or is it dealing with different schema messages? I would add a comment to clarify.